org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/GenerateTestLanguages.mwe2

```diff
@@ -1,63 +1,90 @@
 /*******************************************************************************
- * Copyright (c) 2010 itemis AG (http://www.itemis.eu) and others.
+ * Copyright (c) 2010, 2019 itemis AG (http://www.itemis.eu) and others.
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
  * http://www.eclipse.org/legal/epl-v10.html
  *******************************************************************************/
 module org.eclipse.xtext.xtext.ui.GenerateTestLanguages
+import org.eclipse.xtext.xtext.generator.*
+import org.eclipse.xtext.xtext.generator.model.project.*
 
-import org.eclipse.emf.mwe2.runtime.workflow.Workflow
-import org.eclipse.emf.mwe.utils.*
-import org.eclipse.xtext.*
-import org.eclipse.xtext.generator.*
-import org.eclipse.xtext.ui.generator.*
-import org.eclipse.xtext.generator.ecore2xtext.Ecore2XtextValueConverterServiceFragment
-
-var projectName = "org.eclipse.xtext.xtext.ui.tests"
-var runtimeProject = "../${projectName}"
-var lineDelimiter = '\n'
+var rootPath = ".."
 
 Workflow {
-	bean = StandaloneSetup {
-		ignoreBrokenProjectFiles = true
-		platformUri = "${runtimeProject}/../.."
-		registerEcoreFile = "platform:/resource/org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/ecore2xtext/Ecore2XtextTest.ecore"
-		registerGenModelFile = "platform:/resource/org.eclipse.xtext.xtext.ui.tests/src/org/eclipse/xtext/xtext/ui/ecore2xtext/Ecore2XtextTest.genmodel"
-	} 
-	
-	component = DirectoryCleaner {
-		directory="${runtimeProject}/src-gen"
-	} 
 	
-	component = Generator auto-inject {
-		pathRtProject = "${runtimeProject}"
-		pathUiProject = "${runtimeProject}"
-		projectNameRt = "${projectName}"
-		activator = "org.eclipse.xtext.xtext.ui.tests.Activator"
-		
-		language = { 
-			uri="classpath:/org/eclipse/xtext/xtext/ui/ecore2xtext/Ecore2XtextTest.xtext"
-			fragment = ecore.EcoreGeneratorFragment auto-inject {
-				emfRuntimeVersion = "2.9"
-			}
-			fragment = grammarAccess.GrammarAccessFragment auto-inject {}
-			fragment = resourceFactory.ResourceFactoryFragment {}
-			fragment = generator.serializer.SerializerFragment auto-inject {
-				srcGenOnly = true
-			}
-			fragment = parser.antlr.XtextAntlrGeneratorFragment auto-inject {
-				options = {
-					classSplitting = true
+	component = XtextGenerator {
+		configuration = {
+			project = StandardProjectConfig {
+				baseName = "org.eclipse.xtext.xtext.ui.tests"
+				rootPath = rootPath
+				runtime = {
+					name = "org.eclipse.xtext.xtext.ui.tests"
+				}
+				genericIde = {
+					name = "org.eclipse.xtext.xtext.ui.tests"
+					enabled = true
 				}
+				eclipsePlugin = {
+					name = "org.eclipse.xtext.xtext.ui.tests"
+					enabled = true
+				}
+				createEclipseMetaData = true
+			}
+			code = {
+				encoding = "UTF-8"
```

Comment:
I thought we are using ISO in our code base?

Reply:
will change